From cc1ae4f9d3fc7afb9bd40783d8b97d82910d43a0 Mon Sep 17 00:00:00 2001
From: Tulip Blossom <tulilirockz@proton.me>
Date: Mon, 14 Apr 2025 16:46:46 -0300
Subject: [PATCH 1/2] feat(install): add specific options for RPM
 buildroot-like environments

I wanted to use the install script so I didnt need to handle the config files and systemd units, but I ran into a few issues because the install script kept replacing the DEST_DIR with a location in `%{buildroot}` on the RPM build, this should add a few options to make this more seamless for weird aah enviroments like this.

- `no-pip-build`: You'd build the wheel previously on a package
- `no-override-python-installation-location`: This breaks RPM builds cuz it finds %{BUILDROOT} on the script and it craps itself
---
 install.sh | 51 ++++++++++++++++++++++++++++++++++++++++++++-------
 1 file changed, 44 insertions(+), 7 deletions(-)

diff --git a/install.sh b/install.sh
index 5b166c2..946e2a4 100755
--- a/install.sh
+++ b/install.sh
@@ -3,7 +3,7 @@ set -e
 
 # Argument parsing
 SHORT=r,d:,p:,s:,h
-LONG=remove,dest-dir:,prefix-dir:,sysconf-dir:,no-ectool,no-pre-uninstall,no-post-install,no-battery-sensors,no-sudo,no-pip-install,pipx,help
+LONG=remove,dest-dir:,prefix-dir:,sysconf-dir:,no-ectool,no-pre-uninstall,no-post-install,no-battery-sensors,no-sudo,no-pip-install,no-pip-build,no-override-python-installation-path,pipx,help
 VALID_ARGS=$(getopt -a --options $SHORT --longoptions $LONG -- "$@")
 if [[ $? -ne 0 ]]; then
     exit 1;
@@ -22,6 +22,8 @@ SHOULD_REMOVE=false
 NO_BATTERY_SENSOR=false
 NO_SUDO=false
 NO_PIP_INSTALL=false
+NO_PIP_BUILD=false
+NO_OVERRIDE_PYTHON_INSTALLATION_PATH=false
 DEFAULT_PYTHON_PATH="/usr/bin/python3"
 PIPX=false
 
@@ -58,6 +60,13 @@ while true; do
     '--no-sudo')
         NO_SUDO=true
         ;;
+    '--no-pip-build')
+        NO_PIP_BUILD=true
+        NO_PIP_INSTALL=true
+        ;;
+    '--no-override-python-installation-path')
+        NO_OVERRIDE_PYTHON_INSTALLATION_PATH=true
+        ;;
     '--no-pip-install')
         NO_PIP_INSTALL=true
         ;;
@@ -94,7 +103,7 @@ if [ "$PIPX" = true ]; then
     fi
 fi
 
-if [ "$SHOULD_REMOVE" = false ]; then
+if [ "$SHOULD_REMOVE" = false ] && [ "$NO_PIP_BUILD" = false ]; then
     if ! python -m build -h 1>/dev/null 2>&1; then
         echo "Missing python package 'build'!"
         exit 1
@@ -102,7 +111,6 @@ if [ "$SHOULD_REMOVE" = false ]; then
 fi
 
 PYTHON_SCRIPT_INSTALLATION_PATH="$DEST_DIR$PREFIX_DIR/bin/fw-fanctrl"
-
 # Root check
 if [ "$EUID" -ne 0 ] && [ "$NO_SUDO" = false ]
   then echo "This program requires root permissions or use the '--no-sudo' option"
@@ -196,9 +204,11 @@ function install() {
     fi
     mkdir -p "$DEST_DIR$SYSCONF_DIR/fw-fanctrl"
 
-    build
+    if [ "$NO_PIP_BUILD" = false ] ; then 
+        build
+    fi
 
-    if [ "$NO_PIP_INSTALL" = false ]; then
+    if [ "$NO_PIP_INSTALL" = false ] && [ "$NO_PIP_BUILD" != false ]; then
         echo "installing python package"
         if [ "$PIPX" = false ]; then
             python -m pip install --prefix="$DEST_DIR$PREFIX_DIR" dist/*.tar.gz
@@ -235,7 +245,22 @@ function install() {
             systemctl stop "$SERVICE"
         fi
         echo "creating '$DEST_DIR$PREFIX_DIR/lib/systemd/system/$SERVICE$SERVICE_EXTENSION'"
-        cat "$SERVICES_DIR/$SERVICE$SERVICE_EXTENSION" | sed -e "s,%DEFAULT_PYTHON_PATH%,${DEFAULT_PYTHON_PATH}," | sed -e "s/%PYTHON_SCRIPT_INSTALLATION_PATH%/${PYTHON_SCRIPT_INSTALLATION_PATH//\//\\/}/" | sed -e "s/%SYSCONF_DIRECTORY%/${SYSCONF_DIR//\//\\/}/" | sed -e "s/%NO_BATTERY_SENSOR_OPTION%/${NO_BATTERY_SENSOR_OPTION}/" | tee "$DEST_DIR$PREFIX_DIR/lib/systemd/system/$SERVICE$SERVICE_EXTENSION" > "/dev/null"
+        if [ "$NO_OVERRIDE_PYTHON_INSTALLATION_PATH" = true ] ; then
+            OLD_SYSCONFDIR=$SYSCONFDIR
+            OLD_INSTALL_PATH="${PYTHON_SCRIPT_INSTALLATION_PATH}"
+            PYTHON_SCRIPT_INSTALLATION_PATH="/usr/bin/fw-fanctrl"
+            SYSCONF_DIR="/etc"
+        fi
+        cat "$SERVICES_DIR/$SERVICE$SERVICE_EXTENSION" | \
+            sed -e "s,%DEFAULT_PYTHON_PATH%,${DEFAULT_PYTHON_PATH}," \
+                -e "s/%PYTHON_SCRIPT_INSTALLATION_PATH%/${PYTHON_SCRIPT_INSTALLATION_PATH//\//\\/}/" \
+                -e "s/%SYSCONF_DIRECTORY%/${SYSCONF_DIR//\//\\/}/" \
+                -e "s/%NO_BATTERY_SENSOR_OPTION%/${NO_BATTERY_SENSOR_OPTION}/" | \
+                tee "$DEST_DIR$PREFIX_DIR/lib/systemd/system/$SERVICE$SERVICE_EXTENSION" > "/dev/null"
+        if [ "$NO_OVERRIDE_PYTHON_INSTALLATION_PATH" = true ] ; then
+            SYSCONF_DIR="${OLD_SYSCONFDIR}"
+            PYTHON_SCRIPT_INSTALLATION_PATH="${OLD_INSTALL_PATH}"
+        fi
     done
 
     # add program services sub-configurations based on the sub-configurations present in the './services' folder
@@ -256,7 +281,19 @@ function install() {
         for SUBCONFIG in $SUBCONFIGS ; do
             SUBCONFIG=$(sanitizePath "$SUBCONFIG")
             echo "adding '$DEST_DIR$PREFIX_DIR/lib/systemd/$SERVICE/$SUBCONFIG'"
-            cat "$SERVICES_DIR/$SERVICE/$SUBCONFIG" | sed -e "s/%PYTHON_SCRIPT_INSTALLATION_PATH%/${PYTHON_SCRIPT_INSTALLATION_PATH//\//\\/}/" | tee "$DEST_DIR$PREFIX_DIR/lib/systemd/$SERVICE/$SUBCONFIG" > "/dev/null"
+            if [ "$NO_OVERRIDE_PYTHON_INSTALLATION_PATH" = true ] ; then
+                OLD_SYSCONFDIR=$SYSCONFDIR
+                OLD_INSTALL_PATH="${PYTHON_SCRIPT_INSTALLATION_PATH}"
+                PYTHON_SCRIPT_INSTALLATION_PATH="/usr/bin/fw-fanctrl"
+                SYSCONF_DIR="/etc"
+            fi
+            cat "$SERVICES_DIR/$SERVICE/$SUBCONFIG" | \
+                sed -e "s/%PYTHON_SCRIPT_INSTALLATION_PATH%/${PYTHON_SCRIPT_INSTALLATION_PATH//\//\\/}/" | \
+                tee "$DEST_DIR$PREFIX_DIR/lib/systemd/$SERVICE/$SUBCONFIG" > "/dev/null"
+            if [ "$NO_OVERRIDE_PYTHON_INSTALLATION_PATH" = true ] ; then
+                SYSCONF_DIR="${OLD_SYSCONFDIR}"
+                PYTHON_SCRIPT_INSTALLATION_PATH="${OLD_INSTALL_PATH}"
+            fi
             chmod +x "$DEST_DIR$PREFIX_DIR/lib/systemd/$SERVICE/$SUBCONFIG"
         done
     done

From 9927f91f4a3943f5e881f494b93399fd503bf172 Mon Sep 17 00:00:00 2001
From: Tulip Blossom <tulilirockz@proton.me>
Date: Mon, 14 Apr 2025 17:04:04 -0300
Subject: [PATCH 2/2] fix: make python optional for install phases on packages

---
 install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install.sh b/install.sh
index 946e2a4..bf061ed 100755
--- a/install.sh
+++ b/install.sh
@@ -84,7 +84,7 @@ while true; do
   shift
 done
 
-if ! python -h 1>/dev/null 2>&1; then
+if ! python -h 1>/dev/null 2>&1 && [ "$NO_PIP_BUILD" = false ] ; then
     echo "Missing package 'python'!"
     exit 1
 fi
